gflownet/envs/crystals/composition.py

```diff
@@ -123,7 +123,7 @@ def get_action_space(self):
         assert self.max_diff_elem >= self.min_diff_elem
         assert self.max_atom_i >= self.min_atom_i
         valid_word_len = np.arange(self.min_atom_i, self.max_atom_i + 1)
-        actions = [(element, n) for n in valid_word_len for element in self.elements]
+        actions = [(element, n) for element in self.elements for n in valid_word_len]
```

Comment:
Is this the bug fix?

Reply:
No, but it was a required change for my new reimplementation of composition.get_mask_invalid_actions_forward() which does fix the bug (and it much faster than the original). Having the actions element-contiguous allows get_mask_invalid_actions_forward() to easily set the mask for all actions related to an element all at once and save time in the process.